auction-server/src/auction/service/verification.rs

```
@@ -248,31 +249,63 @@ impl Service {
         }
     }
 
-    fn validate_swap_transaction_instructions(
+    async fn validate_swap_transaction_instructions(
         &self,
         tx: &VersionedTransaction,
+        user_wallet: &Pubkey,
     ) -> Result<(), RestError> {
-        tx.message
-            .instructions()
-            .iter()
-            .enumerate()
-            .try_for_each(|(index, ix)| {
-                self.validate_swap_transaction_instruction(
-                    ix.program_id(tx.message.static_account_keys()),
-                    ix,
-                )
-                .map_err(|e| RestError::InvalidInstruction(Some(index), e))
-            })?;
-
+        for (index, ix) in tx.message.instructions().iter().enumerate() {
+            self.validate_swap_transaction_instruction(
+                ix.program_id(tx.message.static_account_keys()),
+                ix,
+                tx,
+                user_wallet,
+                index,
+            )
+            .await?;
+        }
         Ok(())
     }
 
-    fn validate_swap_transaction_instruction(
+    /// Checks if the instruction provided meets the criteria for validity.
+    /// Instruction must either be an approved program instruction or not contain the user wallet in its accounts.
+    async fn validate_swap_transaction_instruction(
+        &self,
+        program_id: &Pubkey,
+        ix: &CompiledInstruction,
+        tx: &VersionedTransaction,
+        user_wallet: &Pubkey,
+        ix_index: usize,
+    ) -> Result<(), RestError> {
+        if self
+            .check_approved_program_instruction(program_id, ix)
+            .is_ok()
```

it's doesn't feel great that now we swallow the error from `check_approved_program_instruction`. this might make it harder to debug

---

yeah i agree. i refactored this to bubble up the error from `check_approved_program_instruction` if the user account is invoked